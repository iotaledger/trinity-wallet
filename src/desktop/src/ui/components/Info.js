import React from 'react';
import PropTypes from 'prop-types';
import Icon from 'ui/components/Icon';
import css from './info.scss';

/**
 * Info block component
 */
export default class Info extends React.PureComponent {
    static propTypes = {
        /** Info block content */
        children: PropTypes.node.isRequired,
    };

    render() {
        const { children } = this.props;

        return (
            <div className={css.wrapper}>
                <Icon icon="info" size={28} />
                {children}
            </div>
        );
    }
}
