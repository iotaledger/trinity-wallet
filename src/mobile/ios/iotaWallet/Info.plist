<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>BugsnagAPIKey</key>
	<string>fakeAPIkey</string>
	<key>CFBundleAllowMixedLocalizations</key>
	<true/>
	<key>CFBundleDevelopmentRegion</key>
	<string>en</string>
	<key>CFBundleDisplayName</key>
	<string>Trinity</string>
	<key>CFBundleExecutable</key>
	<string>$(EXECUTABLE_NAME)</string>
	<key>CFBundleIdentifier</key>
	<string>$(PRODUCT_BUNDLE_IDENTIFIER)</string>
	<key>CFBundleInfoDictionaryVersion</key>
	<string>6.0</string>
	<key>CFBundleName</key>
	<string>$(PRODUCT_NAME)</string>
	<key>CFBundlePackageType</key>
	<string>APPL</string>
	<key>CFBundleShortVersionString</key>
	<string>0.6.1</string>
	<key>CFBundleSignature</key>
	<string>????</string>
	<key>CFBundleVersion</key>
	<string>40</string>
	<key>ITSAppUsesNonExemptEncryption</key>
	<false/>
	<key>LSApplicationQueriesSchemes</key>
	<array>
		<string>whatsapp</string>
		<string>mailto</string>
	</array>
	<key>LSRequiresIPhoneOS</key>
	<true/>
	<key>NSAppTransportSecurity</key>
	<dict>
		<key>NSAllowsArbitraryLoads</key>
		<false/>
		<key>NSExceptionDomains</key>
		<dict>
			<key>localhost</key>
			<dict>
				<key>NSExceptionAllowsInsecureHTTPLoads</key>
				<true/>
			</dict>
		</dict>
	</dict>
	<key>NSAppleMusicUsageDescription</key>
	<string>Trinity does not require this permission for any functionality. This permission is solely listed in order to satisfy App Store requirements.</string>
	<key>NSBluetoothPeripheralUsageDescription</key>
	<string>Trinity does not require this permission for any functionality. This permission is solely listed in order to satisfy App Store requirements.</string>
	<key>NSCalendarsUsageDescription</key>
	<string>Trinity does not require this permission for any functionality. This permission is solely listed in order to satisfy App Store requirements.</string>
	<key>NSCameraUsageDescription</key>
	<string>Trinity requires access to the camera in order to enable QR scanning.</string>
	<key>NSFaceIDUsageDescription</key>
	<string>Trinity requires access to Face ID in order to enable biometric authentication.</string>
	<key>NSHumanReadableCopyright</key>
	<string>2018 IOTA Foundation. All rights reserved.</string>
	<key>NSLocationWhenInUseUsageDescription</key>
	<string>Trinity does not require this permission for any functionality. This permission is solely listed in order to satisfy App Store requirements.</string>
	<key>NSMotionUsageDescription</key>
	<string>Trinity does not require this permission for any functionality. This permission is solely listed in order to satisfy App Store requirements.</string>
	<key>NSPhotoLibraryAddUsageDescription</key>
	<string>Trinity requires access to your photo library in order to save QR codes to the camera roll.</string>
	<key>NSMicrophoneUsageDescription</key>
	<string>Trinity requires access to your microphone for camera functionality.</string>
	<key>NSPhotoLibraryUsageDescription</key>
	<string>Trinity does not require this permission for any functionality. This permission is solely listed in order to satisfy App Store requirements.</string>
	<key>NSSpeechRecognitionUsageDescription</key>
	<string>Trinity does not require this permission for any functionality. This permission is solely listed in order to satisfy App Store requirements.</string>
	<key>UIAppFonts</key>
	<array>
		<string>SourceSansPro-Bold.ttf</string>
		<string>SourceSansPro-Light.ttf</string>
		<string>SourceSansPro-Regular.ttf</string>
		<string>Inconsolata-Regular.ttf</string>
		<string>Inconsolata-Bold.ttf</string>
		<string>Icons.ttf</string>
		<string>SourceSansPro-SemiBold.ttf</string>
		<string>SourceCodePro-Medium.ttf</string>
	</array>
	<key>UILaunchStoryboardName</key>
	<string>Launch Screen</string>
	<key>UIRequiredDeviceCapabilities</key>
	<array>
		<string>armv7</string>
	</array>
	<key>UIStatusBarHidden</key>
	<true/>
	<key>UIStatusBarStyle</key>
	<string>UIStatusBarStyleLightContent</string>
	<key>UISupportedInterfaceOrientations</key>
	<array>
		<string>UIInterfaceOrientationPortrait</string>
	</array>
	<key>UIViewControllerBasedStatusBarAppearance</key>
	<false/>
</dict>
</plist>
