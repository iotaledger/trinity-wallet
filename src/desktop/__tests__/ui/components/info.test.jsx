import React from 'react';
import { shallow } from 'enzyme';

import Info from 'ui/components/Info';

const props = {
    children: 'Lorem ipsum',
};

describe('Info component', () => {
    test('Render the component', () => {
        const wrapper = shallow(<Info {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Info content', () => {
        const wrapper = shallow(<Info {...props} />);

        expect(wrapper.text()).toEqual('<Icon />Lorem ipsum');
    });
});
