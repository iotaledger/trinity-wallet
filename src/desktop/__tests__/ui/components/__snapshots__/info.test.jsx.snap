// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info component Render the component 1`] = `
<div
  className="wrapper"
>
  <Icon
    icon="info"
    size={28}
  />
  Lorem ipsum
</div>
`;
